src/viam/examples/project/pkg-config/Makefile

```diff
@@ -1,19 +1,24 @@
 CXX = c++
 
-CXXFLAGS = -std=c++14
-CXXFLAGS += `pkg-config --cflags viam-cpp-sdk-libviamsdk`
-LINKFLAGS = `pkg-config --libs viam-cpp-sdk-libviamsdk`
+VIAM_PREFIX ?= ../../../../../build/install
+PKG_CONFIG_PATH ?= $(VIAM_PREFIX)/lib/pkgconfig
+
+
+CXXFLAGS = -std=c++17
+CXXFLAGS += `PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --cflags viam-cpp-sdk-libviamsdk`
+LINKFLAGS = `PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --libs viam-cpp-sdk-libviamsdk`
+LDFLAGS ="-Wl,-rpath,$(VIAM_PREFIX)/"
 
 all: example_dial example_module
 
 example_dial.o: ../../dial/example_dial.cpp
 	$(CXX) $(CXXFLAGS) -c -o $@ $^
 
 example_dial: example_dial.o
-	$(CXX) $(LINKFLAGS) -o $@ $^
+	$(CXX) $(LINKFLAGS) $(LDFLAGS) -o $@ $^
 
-example_module.o: ../../modules/example_module.cpp
+example_module.o: ../../modules/simple/main.cpp
```

Comment:
- Is there some way in make that we can validate that the cwd is actually inside the (a?) `src/viam/examples/project/pkg-config` tree so that these paths are sane?
- We didn't notice that this broke because we don't build these examples in CI. How difficult would it be to fold that into this review?

Reply:
Unfortunately I don't think Makefile has any notion analogous to `CMAKE_CURRENT_SOURCE_DIR`. I poked around and found that sqlite does it by having autoconf set this in a Makefile.ac: https://github.com/sqlite/sqlite/blob/master/Makefile.in#L15-L18

This maybe would be overkill for our sample Makefile though.

Also just for posterity, responding to the second bullet point that CI builds of examples are very much the intended outcome of this task, but we're keeping them in separate tickets for now.